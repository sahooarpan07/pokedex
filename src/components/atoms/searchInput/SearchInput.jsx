import React from 'react';
import {View, TextInput} from 'react-native';
import SearchIcon from 'react-native-vector-icons/EvilIcons';
import {styles} from './SearchInput.styles';
const SearchInput = ({searchText, handleOnChange}) => {
  return (
    <View style={styles.container}>
      <TextInput
        placeholder="Search pokemon here"
        onChangeText={text => {
          handleOnChange(text);
        }}
        style={styles.input}
        value={searchText}
        testID="text-input"
      />
      <View style={styles.icon}>
        <SearchIcon name="search" size={30} color="black" />
      </View>
    </View>
  );
};

export default SearchInput;
