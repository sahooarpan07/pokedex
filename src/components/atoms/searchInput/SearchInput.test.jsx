/**
 * @format
 */

import 'react-native';
import React from 'react';
import SearchInput from './SearchInput';
import {render, fireEvent, screen} from '@testing-library/react-native';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('should render SearchInput Filter', () => {
  const tree = renderer
    .create(<SearchInput searchText={'Bu'} handleOnChange={jest.fn()} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('should display children when expanded', () => {
  const handleOnChange = jest.fn();
  render(<SearchInput handleOnChange={handleOnChange} />);
  const searchInput = screen.getByTestId('text-input');
  fireEvent.changeText(searchInput, 'Pi');
  expect(handleOnChange).toHaveBeenCalledWith('Pi');
});
