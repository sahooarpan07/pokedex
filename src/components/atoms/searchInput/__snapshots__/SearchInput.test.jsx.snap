// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SearchInput Filter 1`] = `
<View
  style={
    {
      "flex": 4,
      "position": "relative",
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="Search pokemon here"
    style={
      {
        "backgroundColor": "#fff",
        "borderRadius": 16,
        "borderWidth": 2,
        "margin": 12,
        "padding": 16,
      }
    }
    testID="text-input"
    value="Bu"
  />
  <View
    style={
      {
        "position": "absolute",
        "right": 30,
        "top": 30,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "black",
            "fontSize": 30,
          },
          undefined,
          {
            "fontFamily": "EvilIcons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      ï„¹
    </Text>
  </View>
</View>
`;
